packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -64,23 +83,32 @@ function NavigationMenuSelector(
 	const hasManagePermissions =
 		canUserCreateNavigationMenu || canUserUpdateNavigationMenu;
 
+	useEffect( () => {
+		if ( ! hasNavigationMenus ) {
+			setSelectorLabel( __( 'No menus. Create one?' ) );
+		} else if ( currentMenuId === null ) {
+			setSelectorLabel( __( 'Select another menu' ) );
+		}
+	}, [ currentMenuId, hasNavigationMenus ] );
+
 	// Show the selector if:
 	// - has switch or create permissions and there are block or classic menus.
 	// - user has create or update permissions and component should show the menu actions.
 	const showSelectMenus =
 		( ( canSwitchNavigationMenu || canUserCreateNavigationMenu ) &&
 			( hasNavigationMenus || hasClassicMenus ) ) ||
-		( hasManagePermissions && showManageActions );
+		hasManagePermissions;
 
 	if ( ! showSelectMenus ) {
 		return null;
 	}
 
 	return (
-		<ToolbarDropdownMenu
+		<DropdownMenu
+			className="wp-block-navigation__navigation-selector"
 			ref={ forwardedRef }
-			label={ __( 'Select Menu' ) }
-			text={ __( 'Select Menu' ) }
+			label={ selectorLabel }
```

Comment:
It's noticeable that there's no visible label for this dropdown and now the aria-label is also the menu name. For screen reader users, there's no way to know what this menu does.

Most controls in the inspector seem to have a label. I did wonder if a different component could be used here, like a custom select control, see the typography tools:
<img width="269" alt="Screen Shot 2022-08-29 at 4 37 44 pm" src="https://user-images.githubusercontent.com/677833/187138489-d8543cf2-b78d-4ab5-a55c-1272c42c84bc.png">

Reply:
I'll try to add a label before trying to change from this dropdown to a custom select.

Reply:
I don't think you can add a label to a dropdown. It has a button to open it and that's it.

Reply:
Would `aria-describedby` work as a prop for the button and then add a hidden descriptor?